setup.py

```diff
@@ -263,12 +263,7 @@ def build_extension(self, ext):
     Extension(
         "pymongo._cmessage",
         include_dirs=["bson"],
-        sources=[
-            "pymongo/_cmessagemodule.c",
-            "bson/_cbsonmodule.c",
-            "bson/time64.c",
-            "bson/buffer.c",
-        ],
```

Comment:
I reverted this change made in PYTHON-3717 because I suspect it was unneeded.

Reply:
Looks like it is: `ImportError: /data/mci/881d33b0b5dec4eb9cb9c1f9be10aa61/src/pymongo/_cmessage.cpython-310-x86_64-linux-gnu.so: undefined symbol: cbson_long_long_to_str`

Reply:
Unless removing `extern` is what caused the error.

Reply:
I'm going to merge the codec options change and open a new ticket to investigate this issue.